internal/flag/error_code.go

```diff
@@ -12,6 +12,7 @@ const (
 	ErrorCodeGeneral             ErrorCode = "GENERAL"
 	ErrorCodeInvalidContext      ErrorCode = "INVALID_CONTEXT"
 	ErrorCodeTargetingKeyMissing ErrorCode = "TARGETING_KEY_MISSING"
+	ErrorCodeBucketingKeyMissing ErrorCode = "BUCKETING_KEY_MISSING"
```

Comment:
I am a bit worried about this one because it is not part of the valid OpenFeature error code _(see list [here](https://openfeature.dev/specification/types/#error-code))._

It means that every Go Feature Flag provider will fall back to `GENERAL` and it will not give any particular information to the user of this flag on why we always use the default value provided.

Reply:
> missing something from the context that is not a targeting key

It is a targeting key though, in this case. I'm tending towards `ErrorCodeTargetingKeyMissing` - looking at the flag definition and noticing it has a `bucketingKey` makes it obvious what happened - if the error says 'invalid context' I think it's a much wider surface of things that could have gone wrong